src/Dapr.Actors/ActorReentrancyConfig.cs

```diff
@@ -0,0 +1,51 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+namespace Dapr.Actors
+{
+    /// <summary>
+    /// Represents the configuration required for Actor Reentrancy.
+    ///
+    /// See: https://docs.dapr.io/developing-applications/building-blocks/actors/actor-reentrancy/
+    /// </summary>
+    public sealed class ActorReentrancyConfig 
+    {
+        private bool enabled;
+        private int maxStackDepth;
+
+        /// <summary>
+        /// Determines if Actor Reentrancy is enabled or disabled.
+        /// </summary>
+        public bool Enabled
+        {
+            get 
+            {
+                return this.enabled;
+            }
+
+            set 
+            {
+                this.enabled = value;
+            }
+        }
+
+        /// <summary>
+        /// Optional parameter that will stop a reentrant call from progressing past the defined
+        /// limit. This is a safety measure against infinite reentrant calls.
+        /// </summary>
+        public int MaxStackDepth
+        {
+            get
+            {
+                return this.maxStackDepth;
+            }
+
+            set
+            {
+                this.maxStackDepth = value;
```

Comment:
What's the default/semantics of this when its value is zero? Does it make sense for this to be `int?` so it defaults to null (unset)

Reply:
If we pass a null through, it assumes a default value on the dapr sidecar. I removed the nullable cause I gave it a default value, but perhaps I was too aggressive in my defaulting?

Reply:
discussed offline, it's better if we can avoid building default values into the SDKs, and rely on the runtime's behavior.